### DISCUSSION TITLE
Error saat fit dengan prophet

### DISCUSSION THREAD

**Discussion by HeinzSiahaan** [teamalgoritma/community#671] [Q&A] [answered]
Saya terkendala error saat melakukan fit dengan prophet

![image](https://github.com/teamalgoritma/community/assets/42597851/d9d97ab9-177c-47b2-8754-f2fde468aca1)

Berikut errornya

21:19:57 - cmdstanpy - INFO - Chain [1] start processing
21:20:03 - cmdstanpy - INFO - Chain [1] done processing
21:20:03 - cmdstanpy - ERROR - Chain [1] error: terminated by signal 3221225657 
Optimization terminated abnormally. Falling back to Newton.
21:20:04 - cmdstanpy - INFO - Chain [1] start processing
21:20:05 - cmdstanpy - INFO - Chain [1] done processing
21:20:05 - cmdstanpy - ERROR - Chain [1] error: terminated by signal 3221225657 

---------------------------------------------------------------------------
```
RuntimeError                              Traceback (most recent call last)
File ~\AppData\Local\r-miniconda\envs\vulcan_da\lib\site-packages\prophet\models.py:96, in CmdStanPyBackend.fit(self, stan_init, stan_data, **kwargs)
     95 try:
---> 96     self.stan_fit = self.model.optimize(**args)
     97 except RuntimeError as e:
     98     # Fall back on Newton

File ~\AppData\Local\r-miniconda\envs\vulcan_da\lib\site-packages\cmdstanpy\model.py:738, in CmdStanModel.optimize(self, data, seed, inits, output_dir, sig_figs, save_profile, algorithm, init_alpha, tol_obj, tol_rel_obj, tol_grad, tol_rel_grad, tol_param, history_size, iter, save_iterations, require_converged, show_console, refresh, time_fmt, timeout)
    737     else:
--> 738         raise RuntimeError(msg)
    739 mle = CmdStanMLE(runset)

RuntimeError: Error during optimization! Command 'C:\Users\user\AppData\Local\r-miniconda\envs\vulcan_da\Lib\site-packages\prophet\stan_model\prophet_model.bin random seed=88527 data file=C:\Users\user\AppData\Local\Temp\tmpo441h7i1\fc_bcleu.json init=C:\Users\user\AppData\Local\Temp\tmpo441h7i1\of5aj1c6.json output file=C:\Users\user\AppData\Local\Temp\tmpo441h7i1\prophet_modelropllpsr\prophet_model-20230524211957.csv method=optimize algorithm=lbfgs iter=10000' failed: 

During handling of the above exception, another exception occurred:

RuntimeError                              Traceback (most recent call last)
Cell In[5], line 2
      1 m = Prophet()
----> 2 m.fit(df)

File ~\AppData\Local\r-miniconda\envs\vulcan_da\lib\site-packages\prophet\forecaster.py:1174, in Prophet.fit(self, df, **kwargs)
   1172     self.params = self.stan_backend.sampling(stan_init, dat, self.mcmc_samples, **kwargs)
   1173 else:
-> 1174     self.params = self.stan_backend.fit(stan_init, dat, **kwargs)
   1176 self.stan_fit = self.stan_backend.stan_fit
   1177 # If no changepoints were requested, replace delta with 0s

File ~\AppData\Local\r-miniconda\envs\vulcan_da\lib\site-packages\prophet\models.py:103, in CmdStanPyBackend.fit(self, stan_init, stan_data, **kwargs)
    101     logger.warning('Optimization terminated abnormally. Falling back to Newton.')
    102     args['algorithm'] = 'Newton'
--> 103     self.stan_fit = self.model.optimize(**args)
    104 params = self.stan_to_dict_numpy(
    105     self.stan_fit.column_names, self.stan_fit.optimized_params_np)
    106 for par in params:

File ~\AppData\Local\r-miniconda\envs\vulcan_da\lib\site-packages\cmdstanpy\model.py:738, in CmdStanModel.optimize(self, data, seed, inits, output_dir, sig_figs, save_profile, algorithm, init_alpha, tol_obj, tol_rel_obj, tol_grad, tol_rel_grad, tol_param, history_size, iter, save_iterations, require_converged, show_console, refresh, time_fmt, timeout)
    736         get_logger().warning(msg)
    737     else:
--> 738         raise RuntimeError(msg)
    739 mle = CmdStanMLE(runset)
    740 return mle

RuntimeError: Error during optimization! Command 'C:\Users\user\AppData\Local\r-miniconda\envs\vulcan_da\Lib\site-packages\prophet\stan_model\prophet_model.bin random seed=67643 data file=C:\Users\user\AppData\Local\Temp\tmpo441h7i1\hmmg0pcf.json init=C:\Users\user\AppData\Local\Temp\tmpo441h7i1\50c720fz.json output file=C:\Users\user\AppData\Local\Temp\tmpo441h7i1\prophet_modelti0_wfx3\prophet_model-20230524212004.csv method=optimize algorithm=newton iter=10000' failed: 
```
Posisi saya sudah ada di pip list
prophet
cmdstanpy
holidays 
dll

Sudah install C++ redistributable untuk VS 2022 juga.

Pada saat saya coba `conda install -c conda-forge prophet` ada pesan seperti di bawah ini. Mungkin perlu menambahkan di variable %PATH% pada Environment Variables. Tapi, saya tidak terlalu mengerti.
![image](https://github.com/teamalgoritma/community/assets/42597851/e080fab7-fa6c-4943-8e28-2b18289a842e)

Mohon bantuannya

**Comment by nadyadtm** [comment]
Dear Pak @HeinzSiahaan 
Apakah sudah mencoba solusi berikut? https://github.com/facebook/prophet/issues/2227#issuecomment-1211008532

**Reply by HeinzSiahaan** [reply]
Dear Kak @nadyadtm 

Sudah saya coba di line 2 (tetapi sudah saya comment). Masih tetap tidak bisa, Kak

**Reply by jafarhf1** [reply]
Dear Pak @HeinzSiahaan, 

Boleh tolong diinfokan pak mengenai versi dari package-package yang digunakan seperti versi prophet dan holidays nya

Terima kasih

**Reply by HeinzSiahaan** [reply]
Berikut pip list yang ada di vulcan_da punya saya.
Saya sudah membuat kernel baru based on requirement Kak Irfan, tapi tetap gagal juga.
anyio==3.6.2
appdirs==1.4.4
argon2-cffi==21.3.0
argon2-cffi-bindings==21.2.0
arrow==1.2.3
asttokens==2.2.1
attrs==23.1.0
backcall==0.2.0
beautifulsoup4==4.12.2
bleach==6.0.0
Bottleneck==1.3.5
certifi==2023.5.7
cffi==1.15.1
charset-normalizer==3.1.0
cmdstanpy==1.1.0
colorama==0.4.6
comm==0.1.2
contourpy==1.0.7
convertdate==2.4.0
cryptography==40.0.2
cycler==0.11.0
debugpy==1.6.7
decorator==5.1.1
defusedxml==0.7.1
ephem==4.1.4
executing==1.2.0
fastjsonschema==2.17.1
fonttools==4.39.3
fqdn==1.5.1
frozendict==2.3.8
hijri-converter==2.3.1
holidays==0.24
html5lib==1.1
idna==3.4
importlib-resources==5.12.0
ipykernel==6.17.1
ipython==8.12.0
ipython-genutils==0.2.0
ipywidgets==8.0.4
isoduration==20.11.0
jedi==0.18.2
Jinja2==3.1.2
joblib==1.2.0
jsonpointer==2.3
jsonschema==4.17.3
jupyter==1.0.0
jupyter_client==8.1.0
jupyter-console==6.6.0
jupyter_core==5.3.0
jupyter-events==0.6.3
jupyter_server==2.3.0
jupyter_server_terminals==0.4.4
jupyterlab-pygments==0.2.2
jupyterlab-widgets==3.0.5
kiwisolver==1.4.4
korean-lunar-calendar==0.3.1
LunarCalendar==0.0.9
lxml==4.9.2
MarkupSafe==2.1.2
matplotlib==3.7.1
matplotlib-inline==0.1.6
mistune==2.0.5
mkl-service==2.4.0
multitasking==0.0.11
nbclassic==0.5.2
nbclient==0.7.2
nbconvert==7.2.9
nbformat==5.8.0
nest-asyncio==1.5.6
notebook==6.5.2
notebook_shim==0.2.2
numexpr==2.8.4
numpy==1.24.2
packaging==23.0
pandas==2.0.0
pandocfilters==1.5.0
parso==0.8.3
patsy==0.5.3
pickleshare==0.7.5
Pillow==9.5.0
pip==23.0.1
platformdirs==3.5.1
plotly==5.14.1
prometheus-client==0.16.0
prompt-toolkit==3.0.38
prophet==1.1.3
psutil==5.9.4
pure-eval==0.2.2
pycparser==2.21
Pygments==2.15.1
PyMeeus==0.5.12
pyparsing==3.0.9
pyrsistent==0.19.3
python-dateutil==2.8.2
python-json-logger==2.0.6
pytz==2023.3
pywin32==305
pywinpty==2.0.10
PyYAML==6.0
pyzmq==25.0.2
qtconsole==5.4.0
QtPy==2.3.0
requests==2.29.0
rfc3339-validator==0.1.4
rfc3986-validator==0.1.1
scikit-learn==1.2.2
scipy==1.10.1
seaborn==0.12.2
Send2Trash==1.8.0
setuptools==65.6.3
setuptools-git==1.2
six==1.16.0
sniffio==1.3.0
soupsieve==2.4.1
stack-data==0.6.2
statsmodels==0.13.5
tenacity==8.2.2
terminado==0.17.1
threadpoolctl==3.1.0
tinycss2==1.2.1
tornado==6.2
tqdm==4.65.0
traitlets==5.9.0
tzdata==2023.3
ujson==5.7.0
uri-template==1.2.0
urllib3==1.26.15
wcwidth==0.2.6
webcolors==1.12
webencodings==0.5.1
websocket-client==1.5.1
wheel==0.38.4
widgetsnbextension==4.0.5
wincertstore==0.2
xarray==2023.5.0
yfinance==0.2.18

**Reply by jafarhf1** [reply, marked as answer]
Dear Pak @HeinzSiahaan,

Terima kasih telah melampirkan pip list dari environment yang Bapak miliki.  Bisa dicoba terlebih dahulu untuk mendowngrade versi dari prophetnya dengan cara uninstall terlebih dahulu package prophet dan holidays kemudian lakukan `pip install prophet==1.1.1`

Semoga membantu

**Comment by kinantytasyao** [comment]
Dear @HeinzSiahaan ,

Untuk menjaga agar forum diskusi berjalan kondusif, kami izin melakukan close pada diskusi ini karena belum ada respon setelah lebih dari 7 hari respon terakhir. Semoga dapat dipahami. Terima kasih.
